### PR TITLE
#369 switch mint functionality in bank module to use checked_add instead of simple add which can cause overflow.

### PR DESCRIPTION
**Pull request by dubbelosix**
couple of places for mint using "+" has been replaced with checked_add to avoid overflow issues

### CODE REVIEW

**Comment by citizen-stig** [comment]
It is a minor note, but I believe that having human readable PR names makes discovery easier. It is hard to get what this PR is about, without manually searching for issue. I mean, now it is easy, since context is fresh, but in 1 months time going through some git history will require couple extra steps to understand what was the purpose of this PR.

File: module-system/module-implementations/sov-bank/tests/mint_test.rs [hunk 1]
```diff
@@ -184,4 +184,21 @@ fn mint_token() {
...
+        minter_address: new_holder.clone(),
+    };
+
+    let minted = bank.call(
+        overflow_mint_message.clone(),
+        &authorized_minter_1_context,
+        &mut working_set,
+    );
+    assert!(minted.is_err());
+    assert_eq!("Overflow", minted.err().unwrap().to_string());
```
Comment (citizen-stig) [inline module-system/module-implementations/sov-bank/tests/mint_test.rs:203]: Can we also check, that total supply hasn't change after that?
Technically it is possible to throw error, but commit changes which affect total supply.

**Comment by bkolad** [comment]
https://github.com/Sovereign-Labs/sovereign-sdk/issues/369

File: module-system/module-implementations/sov-bank/src/token.rs [hunk 2]
```diff
@@ -92,13 +92,18 @@ impl<C: sov_modules_api::Context> Token<C> {
...
-            + amount;
+            .checked_add(amount)
+            .ok_or(anyhow::Error::msg("Overflow"))?;
+
         self.balances.set(minter_address, to_balance, working_set);
-        self.total_supply += amount;
+        self.total_supply = self
+            .total_supply
+            .checked_add(amount)
+            .ok_or(anyhow::Error::msg("Overflow"))?;
```
[the remark this answers is not in the document]
Comment (dubbelosix) [reply module-system/module-implementations/sov-bank/src/token.rs:106]: yup, will do!

[the remark this answers is not in the document]
Comment (dubbelosix) [reply module-system/module-implementations/sov-bank/src/token.rs:106]: yup, just noticed the other comment as well. making the error a bit more verbose

**Review by citizen-stig — APPROVED** [review]

**Pull request merged**